### PR TITLE
Add Xperia 10 III build instructions

### PR DESCRIPTION
**Pull request by sledges**

### CODE REVIEW

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+Without waiting for the middleware to finish, open new shell terminal.
+```nosh
+HABUILD_SDK $
+
+sudo mkdir -p $ANDROID_ROOT-syspart
+sudo chown -R $USER $ANDROID_ROOT-syspart
+cd $ANDROID_ROOT-syspart
+# if you plan to contribute to syspart (/system partition), remove "--depth=1" and "-c" flags below
+repo init -u git://github.com/mer-hybris/android.git -b $HAVERSION -m tagged-manifest.xml --depth=1
```
Comment (voidanix) [inline Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:111]: You can move the `-c` from the `repo sync` below to here.
```suggestion
repo init -u git://github.com/mer-hybris/android.git -b $HAVERSION -m tagged-manifest.xml --depth=1 -c
```

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
+---
+title: Sailfish X Xperia Android 11 Build and Flash
+permalink: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_10_Build_and_Flash/
```
Comment (voidanix) [inline Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:3]:
```suggestion
permalink: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/
```

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+sudo apt-get install libssl-dev
+sudo mkdir -p $ANDROID_ROOT
+sudo chown -R $USER $ANDROID_ROOT
+cd $ANDROID_ROOT
+git clone --recurse-submodules https://github.com/mer-hybris/droid-hal-sony-$FAMILY .
+
+git config --global user.name "Your Name"
+git config --global user.email "you@example.com"
+
+repo init -u git://github.com/mer-hybris/android.git -b $HAVERSION -m tagged-localbuild.xml
```
Comment (voidanix) [inline Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:56]: Can you add `--depth=1 -c` here like with `hadk-syspart` along with the comment for contributions?

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+cd $ANDROID_ROOT
+sudo zypper ref
+rpm/dhd/helpers/build_packages.sh --droid-hal
+git clone --recursive https://github.com/mer-hybris/droid-config-sony-$FAMILY hybris/droid-configs -b master
+if [ -z "$(grep patterns-sailfish-consumer-generic $ANDROID_ROOT/hybris/droid-configs/patterns/patterns-sailfish-device-configuration-$DEVICE.inc)" ]; then
+  sed -i "/Summary: Jolla Configuration $DEVICE/aRequires: patterns-sailfish-consumer-generic\n\n# Early stages of porting benefit from these:\nRequires: patterns-sailfish-device-tools" $ANDROID_ROOT/hybris/droid-configs/patterns/patterns-sailfish-device-configuration-$DEVICE.inc
+fi
+if [ -z "$(grep jolla-devicelock-daemon-encsfa $ANDROID_ROOT/hybris/droid-configs/patterns/patterns-sailfish-device-adaptation-$HABUILD_DEVICE.inc)" ]; then
+  sed -i "s/sailfish-devicelock-fpd/jolla-devicelock-daemon-encsfa/" $ANDROID_ROOT/hybris/droid-configs/patterns/patterns-sailfish-device-adaptation-$HABUILD_DEVICE.inc
```
Comment (voidanix) [inline Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:93]: May I ask what is the point of this? Shouldn't it be the opposite like below?
```suggestion
  sed -i "s/jolla-devicelock-daemon-encsfa/sailfish-devicelock-fpd/" $ANDROID_ROOT/hybris/droid-configs/patterns/patterns-sailfish-device-adaptation-$HABUILD_DEVICE.inc
```

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+```
+
+You can now close the above shell terminal. Next, please proceed with:
+```nosh
+PLATFORM_SDK $
+
+cd $ANDROID_ROOT
+rpm/dhd/helpers/build_packages.sh --gg
+rpm/dhd/helpers/build_bootimg_packages.sh
+git clone --recursive https://github.com/mer-hybris/droid-hal-img-boot-sony-$FAMILY hybris/mw/droid-hal-img-boot-sony-$FAMILY
```
Comment (voidanix) [inline Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:158]: This doesn't seem necessary as the line below will also pull the submodule with it

**Review by voidanix — COMMENTED** [review]
Just a few nitpicks, the rest looks OK

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md (hunk 4 above)
Comment (voidanix) [reply Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:93]: I'm assuming this has something to do with `sailfish-fpd-slave`, even though I thought it was going to get pulled by mic.
Feel free to resolve this thread.

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+Please download the latest Sailfish OS HADK (Hardware Adaptation Development Kit) from within [this link](https://sailfishos.org/hadk).
+
+Please check the requirements for your build host: <https://source.android.com/setup/build/requirements>
+
+Minimum Sailfish OS version for this port is 4.3.0.
+
+If you are new to HADK, please carefully read the disclaimer on page 1, then **chapters 1 and 2**.
+
+The disk space requirement for this build is not what HADK says, but around 250GB . The download size requirement is around 50GB.
```
[the remark this answers is not in the document]
Comment (sledges) [reply Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:27]: `-syspart` says 142G (.repo within reports 17G of downloads). Main dhd checkout is less than that, so 250G should be within ballpark

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+
+HADK uses LineageOS/CyanogenMod as reference base. Here we'll instead have AOSP (Android Open Source Project) 11.0. Now you can read through **chapter 3** of the HADK.
+
+If you ever run into difficulties, please look through the links in the topic of the [#sailfishos-porters](https://webchat.oftc.net/?channels=#sailfishos-porters) channel on IRC at oftc.net.
+
+In **chapter 4** (*Setting up the SDKs*) setup the environment as requested in ~/.hadk.env, but then set and add the following additional variables (here we'll build for Sony Xperia 10 III, Dual SIM variant being its only one):
```
[the remark this answers is not in the document]
Comment (sledges) [reply Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:33]: Many porters prefer to set their own values for directories, having all in one bunch will neglect readers attention from the actual PDF, i.e. the concept of arch is also important to understand, as opposed to just copy-pasting:)

**Comment by sledges** [comment]
This is now ready for final review (the build was broken and required fixes to build_package.sh which now have been merged).

File: Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Thaodan) [reply Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md:27]: It depends,  for me syspart and hybris hal android clones together are like 350GB (343,5 GiB (368.877.240.854)).

**Comment by Thaodan** [comment]
Looks fine for me except the size of the downloaded/build sources can vary.

**Review by Thaodan — APPROVED** [review]

**Pull request merged**